Registration.jsx

```diff
@@ -0,0 +1,170 @@
+// global
+import React, { useState, useEffect } from 'react';
+
+import axios from 'axios';
+// components
+import {Button, Input} from 'blueprint-react';
+import { useHistory } from 'react-router-dom'
+import { Card, CardHeader, CardBody, Icon } from '@bui/magnetic';
+// styles
+import './registration.scss';
+import _ from 'lodash';
+// import { response } from 'express';
+import jwt_decode from "jwt-decode";
+
+
+function Register() {
+    const history = useHistory();
+
+    const [user, setUser] = useState(null);
+    const [username, setUsername] = useState("");
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [passwordConfirmation, setPasswordConfirmation] = useState("");
+
+
+    const refreshToken = async () => {
+      try {
+        const res = await axios.post("/refresh", { token: user.refreshToken });
+        setUser({
+          ...user,
+          accessToken: res.data.accessToken,
+          refreshToken: res.data.refreshToken,
+        });
+        return res.data;
+      } catch (err) {
+        console.log(err);
+      }
+    };
+
+    const axiosJWT = axios.create()
+
+    axiosJWT.interceptors.request.use(
+      async (config) => {
+        let currentDate = new Date();
+        const decodedToken = jwt_decode(user.accessToken);
```

Comment:
instead of decoding token on every request, you can have a periodic refresh after login. All token-related functions can be in one js file